llvm/lib/Target/AArch64/AArch64SVEInstrInfo.td

```diff
@@ -2993,6 +2993,22 @@ let Predicates = [HasSVE_or_SME] in {
   defm : unpred_loadstore_bitcast<nxv2i64>;
   defm : unpred_loadstore_bitcast<nxv2f64>;
 
+  // Allow using LDR/STR to avoid the predicate dependence.
+  let Predicates = [IsLE, AllowMisalignedMemAccesses] in
+    foreach Ty = [ nxv16i8, nxv8i16, nxv4i32, nxv2i64, nxv8f16, nxv4f32, nxv2f64, nxv8bf16 ] in {
+      let AddedComplexity = 2 in {
+        def : Pat<(Ty (load (am_sve_indexed_s9 GPR64sp:$base, simm9:$offset))),
+                  (LDR_ZXI GPR64sp:$base, simm9:$offset)>;
+        def : Pat<(store Ty:$val, (am_sve_indexed_s9 GPR64sp:$base, simm9:$offset)),
+                  (STR_ZXI ZPR:$val, GPR64sp:$base, simm9:$offset)>;
+      }
```

Comment:
Can this live inside `unpred_loadstore_bitcast` because it will be nicer to have all the "unconventional" results of normal loads and stores together. This will mean we have a couple of duplicate patterns for the `nxv16i8` type but I think that should be ok?

Reply:
Thanks, that's where I had them initially, but it seems the predicates weren't being applied when the patterns were in `unpred_loadstore_bitcast`. For example, I was already using the `IsLE` predicate when I opened this PR, but it only became effective when I moved the patterns out into the separate loop as you can [see in the latest commit](https://github.com/llvm/llvm-project/pull/127837/commits/6b1ad6758cd854bc8d15e07b4dae4f2936c416bb#diff-e59a1dfcf45e1736fe516d49e304f65e921ac474ef443b78b2e6ca27abbded68). Although, now that I'm looking at it, the original patterns in `unpred_loadstore_bitcast` also [don't seem to be correct](https://godbolt.org/z/1bvndses7)? Am I missing anything?

Reply:
Ah, I see! Thanks very much, that makes sense! What if I absorb the current patterns into the loop so that we still have unconventional loads/stores grouped together, and add `HasSVE_or_SME` (from the parent definition) to the predicates? Or do you have a better suggestion?